lib/puppet/pops/types/types.rb

```diff
@@ -695,7 +695,7 @@ def self.register_ptype(loader, ir)
   end
 
   def instance?(o, guard = nil)
-    if o.is_a?(String) || o.is_a?(Numeric) || o.is_a?(TrueClass) || o.is_a?(FalseClass) || o.is_a?(Regexp)
+    if o.instance_of?(String) || o.instance_of?(Numeric) || o.instance_of?(TrueClass) || o.instance_of?(FalseClass) || o.instance_of?(Regexp)
```

Comment:
This current implementation solves the issue, although it produces a lot of warnings. Not sure what to do about that

Reply:
Maybe create a single commit that only changes `is_a?` to `instance_of` for string, numeric, ..., Array, Hash and make sure tests pass before going deeper.